opensearch/src/main/java/org/opensearch/sql/opensearch/request/PushDownRequestBuilder.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.request;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import lombok.Getter;
+import org.apache.commons.lang3.tuple.Pair;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.search.aggregations.AggregationBuilder;
+import org.opensearch.search.sort.SortBuilder;
+import org.opensearch.sql.ast.expression.Literal;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.expression.ReferenceExpression;
+import org.opensearch.sql.opensearch.data.type.OpenSearchDataType;
+import org.opensearch.sql.opensearch.data.value.OpenSearchExprValueFactory;
+import org.opensearch.sql.opensearch.response.agg.OpenSearchAggregationResponseParser;
+
+public interface PushDownRequestBuilder {
+
+  default boolean isBoolFilterQuery(QueryBuilder current) {
+    return (current instanceof BoolQueryBuilder);
+  }
+
+  private String throwUnsupported(String operation) {
+    return String.format("%s: push down %s in cursor requests is not supported",
+        getClass().getSimpleName(), operation);
+  }
+
+  default void pushDownFilter(QueryBuilder query) {
+    throw new UnsupportedOperationException(throwUnsupported("filter"));
+  }
```

Comment:
Should we remove all these default implementation? I think if a subclass cannot support one of them, it can do this itself?

Reply:
Fixed in bc8c73fe5.